### PR TITLE
Fix broken form getMessages

### PR DESCRIPTION
**Pull request by mbrostami**
Hello!

* Type: bug fix 
* Link to issue: #13294 

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/cphalcon/blob/master/CONTRIBUTING.md)?
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I wrote some tests for this PR.

Small description of change:

Thanks

### CODE REVIEW

**Comment by mbrostami** [comment]
Will check it out soon.

**Comment by sergeyklay** [comment]
@mbrostami Could you please update change log too.

File: phalcon/forms/form.zep [hunk 1]
```diff
@@ -370,10 +370,23 @@ class Form extends Injectable implements \Countable, \Iterator
 	 */
 	public function getMessages(boolean byItemName = false) -> <Group>
```
Comment (sergeyklay) [inline phalcon/forms/form.zep]:
```zep
 	public function getMessages(boolean byItemName = false) -> <Group> | array
```

Comment (sergeyklay) [reply phalcon/forms/form.zep]: Could you please amend PHPDoc by adding examples from the https://github.com/phalcon/cphalcon/issues/13294 issue

File: phalcon/forms/form.zep [hunk 2]
```diff
@@ -370,10 +370,23 @@ class Form extends Injectable implements \Countable, \Iterator
...
 	{
-		var messages;
+		var messages, messagesByItem, elementMessage;
 
 		let messages = this->_messages;
 		if typeof messages == "object" && messages instanceof Group {
+		    /**
+		     * This part of code is for backward compatibility, it should be removed in next major version
+		     */
+		    if byItemName {
```
Comment (sergeyklay) [inline phalcon/forms/form.zep]:
```zep
if unlikely byItemName
```

For more see: https://stackoverflow.com/a/11227902/1661465

File: phalcon/forms/form.zep [hunk 3]
```diff
@@ -370,10 +370,23 @@ class Form extends Injectable implements \Countable, \Iterator
...
+                let messagesByItem = [];
+                for elementMessage in messages {
+                    if !isset messagesByItem[elementMessages->getField()] {
```
Comment (sergeyklay) [inline phalcon/forms/form.zep]: it seems to me `if fetch` may be faster

**Review by sergeyklay — CHANGES_REQUESTED** [review]

File: phalcon/forms/form.zep (hunk 3 above)
Comment (sergeyklay) [reply phalcon/forms/form.zep]: anyway here is typo `elementMessage` not `elementMessages`

Comment (mbrostami) [reply phalcon/forms/form.zep]: I couldn't find the way to use ```if fetch```, refer to this link https://docs.zephir-lang.com/en/latest/operators.html#fetch I thought that ```isset``` is faster than ```fetch```, please let me know if I'm wrong.

**Comment by mbrostami** [comment]
@sergeyklay Done.

**Review by sergeyklay — APPROVED** [review]

**Comment by sergeyklay** [comment]
@mbrostami Thank you for quick reply and fixes

**Pull request merged**